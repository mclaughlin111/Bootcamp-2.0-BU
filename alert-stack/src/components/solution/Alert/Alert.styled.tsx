import styled from "styled-components";

export const StyledAlert = styled.div`
  color: black;
  border-radius: 8px;

  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
  background-color: lightgray;
`;

export const StyledCloseButton = styled.svg`
  width: 24px;
  padding-left: 20px;
  cursor: pointer;
`;
