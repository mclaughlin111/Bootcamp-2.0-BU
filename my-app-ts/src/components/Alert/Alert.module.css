.alert {
  color: black;
  border-radius: 8px;
  padding: 20px;
}

.info {
  background-color: lightblue;
}

.error {
  background-color: red;
}

.warning {
  background-color: orange;
}

.success {
  background-color: lightgreen;
}
