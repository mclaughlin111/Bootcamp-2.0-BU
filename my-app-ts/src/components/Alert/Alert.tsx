import styles from "./Alert.module.css";
import classNames from "classnames";

export type AlertType = "error" | "warning" | "info" | "success";

type AlertProps = {
  message: string;
  type?: AlertType;
};

export const Alert = ({ message, type = "info" }: AlertProps) => (
  <div className={classNames(styles.alert, styles[type])}>{message}</div>
);
