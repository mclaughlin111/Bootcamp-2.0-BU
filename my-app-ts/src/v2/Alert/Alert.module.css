.alert {
  color: black;
  border-radius: 8px;
  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
}

.alert svg {
  width: 24px;
  padding-left: 20px;
}

.close {
  cursor: pointer;
}

.info {
  background-color: lightblue;
}

.error {
  background-color: red;
}

.warning {
  background-color: orange;
}

.success {
  background-color: lightgreen;
}
