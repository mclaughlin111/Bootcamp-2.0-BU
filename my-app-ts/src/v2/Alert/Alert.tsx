/* eslint-disable jsx-a11y/anchor-is-valid */
import styles from "./Alert.module.css";
import classNames from "classnames";

export type AlertType = "error" | "warning" | "info" | "success";

type AlertProps = {
  message: string;
  type?: AlertType;
};

export const Alert = ({ message, type = "info" }: AlertProps) => {
  return (
    <div className={classNames(styles.alert, styles[type])}>
      <div>{message}</div>
      <div
        className={styles.close}
        onClick={() => console.log("clicked Close")}
      >
        <svg
          fill="black"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 1792 1792"
        >
          <path
            d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4
   c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1
   c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
          />
        </svg>
      </div>
    </div>
  );
};
