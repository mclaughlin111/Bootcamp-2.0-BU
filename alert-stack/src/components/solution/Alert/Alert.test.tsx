import userEvent from "@testing-library/user-event";
import { Alert, AlertType } from "./Alert";
import { render, screen } from "@testing-library/react";
import "jest-styled-components";

test.each<AlertType | undefined>([
  "error",
  "warning",
  "info",
  "success",
  undefined,
])("render test %s", (alertType) => {
  const testMessage = "This is my message now";
  const { container } = render(
    <Alert id={1} message={testMessage} type={alertType} />
  );

  expect(container).toMatchSnapshot();
});

test("Onclosed is fired corectly", async () => {
  const closedMock = jest.fn();

  render(
    <Alert id={8} message={"Another test message"} onClosed={closedMock} />
  );

  const button = screen.getByRole("button");
  await userEvent.click(button);

  expect(closedMock).toHaveBeenCalledTimes(1);
  expect(closedMock).toHaveBeenNthCalledWith(1, 8);
});
