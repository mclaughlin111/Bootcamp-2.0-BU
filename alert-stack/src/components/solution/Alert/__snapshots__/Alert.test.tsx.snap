// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test error 1`] = `
.c0 {
  color: black;
  border-radius: 8px;
  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
  background-color: lightgray;
}

.c1 {
  width: 24px;
  padding-left: 20px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
    role="alert"
  >
    <div>
      This is my message now
    </div>
    <div
      role="button"
    >
      <svg
        class="c1"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`render test info 1`] = `
.c0 {
  color: black;
  border-radius: 8px;
  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
  background-color: lightgray;
}

.c1 {
  width: 24px;
  padding-left: 20px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
    role="alert"
  >
    <div>
      This is my message now
    </div>
    <div
      role="button"
    >
      <svg
        class="c1"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`render test success 1`] = `
.c0 {
  color: black;
  border-radius: 8px;
  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
  background-color: lightgray;
}

.c1 {
  width: 24px;
  padding-left: 20px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
    role="alert"
  >
    <div>
      This is my message now
    </div>
    <div
      role="button"
    >
      <svg
        class="c1"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`render test undefined 1`] = `
.c0 {
  color: black;
  border-radius: 8px;
  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
  background-color: lightgray;
}

.c1 {
  width: 24px;
  padding-left: 20px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
    role="alert"
  >
    <div>
      This is my message now
    </div>
    <div
      role="button"
    >
      <svg
        class="c1"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`render test warning 1`] = `
.c0 {
  color: black;
  border-radius: 8px;
  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
  background-color: lightgray;
}

.c1 {
  width: 24px;
  padding-left: 20px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
    role="alert"
  >
    <div>
      This is my message now
    </div>
    <div
      role="button"
    >
      <svg
        class="c1"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
</div>
`;
